### PR TITLE
Feature/play button and timer

### PR DESCRIPTION
**Pull request by Mengzhen1992**
please check this feature: 
- add play button for each ongoing task
- add countdown timer when the play button is clicked

### CODE REVIEW

File: pages/timer/[id].js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    const hour =
+      Math.floor(durationClock.sec / 3600) < 10
+        ? "0" + Math.floor(durationClock.sec / 3600)
+        : Math.floor(durationClock.sec / 3600);
+    const minutes =
+      durationClock.sec % 3600 === 0
+        ? "00"
+        : Math.floor(durationClock.sec / 3600) >= 1
+        ? Math.floor((durationClock.sec % 3600) / 60)
+        : Math.floor(durationClock.sec / 60) < 10
```
Comment (Mark-Krensel) [inline pages/timer/[id].js]: hm, I think nested terniaries are not the best idea. I would recommend switching to something else, like if else or switch

File: components/TaskItem.js [hunk 2]
```diff
@@ -12,6 +14,14 @@ const TaskItem = ({
   handleToggleTask,
   handleDelete,
 }) => {
+  const router = useRouter();
+  function durationDisplay() {
+    const hourDisplay = duration / 3600 < 1 ? "" : duration / 3600;
+    const minDisplay = duration % 3600 === 0 ? "" : duration / 60;
+    if (hourDisplay === "" && minDisplay === "") return "0min";
+    if (hourDisplay === "") return `${minDisplay}min`;
```
Comment (Mark-Krensel) [inline components/TaskItem.js]: i think you could use else if for the 2nd and 3rd if here

File: pages/timer/[id].js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  useEffect(() => {
+    setDurationClock({ sec: duration, start: true });
+  }, []);
+
+  useEffect(() => {
+    let timer;
+    if (durationClock !== null) {
+      timer = setInterval(() => {
+        if (durationClock.sec > 0) {
```
Comment (Mark-Krensel) [inline pages/timer/[id].js]: also else if here I think

File: pages/timer/[id].js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        : Math.floor(durationClock.sec / 3600) >= 1
+        ? Math.floor((durationClock.sec % 3600) / 60)
+        : Math.floor(durationClock.sec / 60) < 10
+        ? "0" + Math.floor(durationClock.sec / 60)
+        : Math.floor(durationClock.sec / 60);
+    const seconds =
+      durationClock.sec % 60 < 10
+        ? "0" + (durationClock.sec % 60)
+        : durationClock.sec % 60;
+    return `${hour}:${minutes}:${seconds}`;
```
Comment (s-kond) [inline pages/timer/[id].js]: When I tested it, it always returned "NaN:NaN:NaN" - maybe it takes a number as a string somewhere? Unfortunately I couldn't find the mistake...

File: pages/timer/[id].js (hunk 3 above)
Comment (Mengzhen1992) [reply pages/timer/[id].js]: @Mark-Krensel Thank you very much for your suggestions! I did some code refactoring. Would you like to have a look again?

File: pages/timer/[id].js (hunk 4 above)
Comment (Mengzhen1992) [reply pages/timer/[id].js]: @s-kond Hi Stefan i am very happy to receive your feedback about the test of my app. I did some code improvements. Would you like to test it again. Thanks in advance! Preview: https://capstone-project-q7h710i74-mengzhen1992.vercel.app/

File: pages/timer/[id].js [hunk 5]
```diff
@@ -2,60 +2,35 @@ import { useRouter } from "next/router";
...
-        if (durationClock.sec > 0) {
-          setDurationClock((pre) => ({ sec: pre.sec - 1, start: pre.start }));
-        }
-      }, 1000);
-    }
+    let interval = setInterval(() => {
+      if (timer.sec > 0) {
+        setTimer((pre) => ({ sec: pre.sec - 1, start: pre.start }));
+      }
+    }, 1000);
```
Comment (Mark-Krensel) [inline pages/timer/[id].js:26]: this looks good!

File: ultils/index.js [hunk 6]
```diff
@@ -65,3 +65,14 @@ function formatTime(seconds) {
...
+export function displayDuration(seconds) {
+  const { hour, min } = formatTime(seconds);
+  if (hour === 0 && min === 0) {
+    return "0min";
+  } else if (hour === 0) {
+    return `${min}min`;
+  } else if (min === 0) {
+    return `${hour}h`;
+  } else return `${hour}h${min}min`;
+}
```
Comment (Mark-Krensel) [inline ultils/index.js:76]: nice! the structure is a lot cleaner now <3

File: ultils/index.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  ];
+  const day = date.getDay();
+  return year + "." + month + "." + dates + " " + arr[day];
+}
+
+/**
+ *
+ * @returns a string with time format 20:10:36
+ */
+export function displayTime(seconds) {
```
Comment (Mark-Krensel) [inline ultils/index.js:31]: please also check replace the 2nd & 3rd if in this function with an if else

**Review by Mark-Krensel — APPROVED** [review]
left one more suggestion, but otherwise looking good to me :)

File: ultils/index.js (hunk 7 above)
Comment (Mengzhen1992) [reply ultils/index.js:31]: @Mark-Krensel already changed. Thanks again 👍

Comment (Mengzhen1992) [reply ultils/index.js:31]: @Mark-Krensel Hi Mark, i I tried to use if else statement, but something happened when the time was displayed. I think the three if conditions are parallel and can be met at the same time, when man uses else if , the code will go to a condition, once met and will return.

**Pull request merged**